lib/kube/proxy/forwarder.go

```diff
@@ -772,22 +861,42 @@ func (f *Forwarder) authorize(ctx context.Context, actx *authContext) error {
 
 	mfaParams := actx.MFAParams(ap.GetRequireMFAType())
 
+	clusterNotFound := trace.AccessDenied("kubernetes cluster %q not found", actx.kubeCluster)
```

Comment:
It's alright to prepare the message beforehand, but save the actual trace.AccessDenied call to the return statement, so we capture the correct line in the error.